[PORT] Smooth Doors from Yogstation

MNarath1 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This PR ports https://github.com/yogstation13/Yogstation/pull/8901 from monster860 it adds a smoother door animation to every airlock and i guess you could say that i am partially porting https://github.com/yogstation13/Yogstation/pull/9705 trough i only add a cut_overlays().

https://streamable.com/927o6y (outdated)Some airlocks i put there that show some of the new airlock movements
https://streamable.com/w3pum7 a bit speeded up

## Why It's Good For The Game

Smoother Door animation

## Changelog
:cl: 
tweak: Doors are smoother
/:cl:
<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

MNarath1 committed:
Port+somefixes

ghost commented:
This kind of looks cursed not gonna lie, good PR.

ike709 commented:
I feel like it's _slightly_ too slow, but otherwise it looks good.

MNarath1 commented:
> I feel like it's _slightly_ too slow, but otherwise it looks good.

i will see what i can do about the speed i think i got an idea how to fix this and make it a bit  faster

MNarath1 committed:
Speeds up animation and fixes some stuff

speeded up to 3.5 and adds a cut_overlays() to make sure access indicator overlayer don't stack also fixes some stuff with shuttle airlocks

MNarath1 commented:
not sure if i really have to link the PR for a such minor change like adding a cut_overlays() but i am letting it stay for now

PowerfulBacon commented:
this looks absolutely cursed, does this get shutters like in cargo too?

PowerfulBacon commented:
holy shit the alien door is sexy

MNarath1 commented:
> holy shit the alien door is sexy

is the abductor airlock even used anywhere in the game?

MNarath1 commented:
ok scrap that it is on a single emergency shuttle i think

MNarath1 commented:
> this looks absolutely cursed, does this get shutters like in cargo too?

I give in the slower version looked pretty cursed but i think after the speed up it looks pretty good

MNarath1 commented:
Also tell me if you think its too fast now or if its fine

Chessus commented:
Looks lovely!

ike709 commented:
Looks good.

MNarath1 committed:
fixes notes pin overlayer

Chessus commented:
I'm going to sound like a noob but are door opening  animation purely cosmetic? Or when the animation is playing the code says no player can pass?

MNarath1 commented:
> I'm going to sound like a noob but are door opening animation purely cosmetic? Or when the animation is playing the code says no player can pass?

from what i can tell the moment you can pass is determinated when the density var is set to False this seems to be handled by some sleep() that delay the activation of the next part

MNarath1 commented:
actually should i change the sleep() var so that it matches up more with the new animation time or not? at the moment it would be around 1.5 deciseconds off

MNarath1 commented:
the difference is very small but still kinda noticable i think might just imagine it its still just 1.5 deciseconds so this would buff how fast you can go trough doors but only by like 0.15 seconds so yeah

MNarath1 committed:
change the sleep() from 4 to 2.5 to match the animation more

MNarath1 commented:
yeah i think its better that way seems like it was also setup to match the old animation speed before so doing that might be good

ike709 commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -487,144 +528,116 @@
-	var/mutable_appearance/sparks_overlay
-	var/mutable_appearance/note_overlay
-	var/notetype = note_type()
+	for(var/obj/effect/overlay/airlock_part/part in part_overlays)
```
```suggestion
	for(var/obj/effect/overlay/airlock_part/part as() in part_overlays)
```

MNarath1 replied:
implemented

ike709 commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -1165,7 +1182,7 @@
 	set_opacity(0)
 	update_freelook_sight()
-	sleep(4)
+	sleep(2.5)
```
You also need to change the sleep in `close()`. Might be a good idea to replace the values with a var on the airlock so people don't forget about this if we tweak it in the future.

MNarath1 replied:
var for both sleep in open() and close() also implemented

ike709 commented on code/game/objects/effects/overlays.dm:
```diff
@@ -52,6 +52,23 @@
+	var/move_end_time = 3.5
+	var/aperture_angle = 0
+	var/obj/machinery/door/airlock/parent
+// in case some caveman is still using 512
```
You can't compile bee on 512 nor play bee on 512, can this just be removed?

MNarath1 replied:
Removed the unneeded code for 512 support

MNarath1 committed:
Implement suggested changes and fixes

Removing the unneeded support for 512 implementing some suggested changes by ike and fixing a missing airlock icon bug for the external_arrival airlocks

Chessus commented:
Just tested it and looks great. You zoom through fire doors also

MNarath1 commented:
> Just tested it and looks great. You zoom through fire doors also

well the firelocks are not influence by this PR the firelocks are as the airlocks subtypes of doors but they are not subtypes of airlocks so changes at airlocks should not influence them so this is not done by my this PR

MNarath1 commented:
Just to make sure i tested it on master and firelocks are exactly as fast

Chessus commented:
Ah I see, must just be becuase im on a local sever and there's no laggg

MNarath1 committed:
Removes an unneeded var and uses an already existing instead

Making an entire new var seems silly if i can just override an already existing var that actually is for this very reason instead
